money.go

```diff
@@ -105,14 +122,32 @@ func alignRoundingMode(m *Money, ma []*Money) MoneyOption {
 	return WithRoundingMode(RoundBankers)
 }
 
+func alignSmallestDenomination(m *Money, ma []*Money) MoneyOption {
+	if m.smallestDenomination != 0 {
+		return WithSmallestDenomination(m.smallestDenomination)
+	}
+	fm, isFound := lo.Find(ma, func(money *Money) bool {
+		return money.smallestDenomination != 0
```

Comment:
I think it's possibly to have smallest denomination as 0 for some currencies

Reply:
the smallest denomination should never be zero because it's the smallest unit of a currency.